presto-main/src/main/java/com/facebook/presto/server/SimpleHttpResponseHandler.java

```diff
@@ -57,9 +58,14 @@ else if (response.getStatusCode() == HttpStatus.SERVICE_UNAVAILABLE.code()) {
                 callback.failed(new ServiceUnavailableException(uri));
             }
             else {
-                // Something is broken in the server or the client, so fail immediately (includes 500 errors)
                 Exception cause = response.getException();
-                if (cause == null) {
+                String responseErrorCode = response.getHeader("Error-Code");
```

Comment:
Maybe add a comment to explain where/how the header is constructed? BTW, do we wanna make it more presto-specific? For example "Presto-Error-Code". "Error-Code" sounds like a thing that would collide with HTTP standard.

Reply:
Yeah, was talking about this with Masha in another diff's review. I see there are some X-Presto headers in protocol. Do we want to make this header in the protocol as well?

Reply:
Talked with @mbasmanova and we think it should be X-Presto-Error-Code.